extend illegal instruction docs

milachae opened the pull request:
Closes #535

boegel commented on mkdocs/docs/HPC/multi_job_submission.md:
```diff
@@ -138,6 +138,14 @@ program will now be run for all 100 parameter instances -- 8
 
+!!! warning
+    When you want to submit a job on a cluster that is not the default cluster. You need to use:
+    <pre><code>$ <b>module swap env/slurm/<submit cluster\></b>
```
Don't use `<...>`, it will confuse people.

Do something like:

```
For example, to submit a worker job to the [donphan debug cluster](...) from the login nodes, use:
...

milachae replied:
Changed this everywhere to an example (donphan).

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+When using multi-job submissions, it could be possible you get following error when you try to run it on a non-default cluster.
+
+<pre><code>$ <b> wsub</b>
+/apps/gent/RHEL8/skylake-ib/software/worker/1.6.13-iimpi-2021b/bin/wsub: line 27: 2152510 <b>Illegal instruction</b>     (core dumped) ${PERL} ${DIR}/../lib/wsub.pl "$@"
```
Avoid using a specific version (or even OS/CPU arch), since this will go out of data in no time:

```suggestion
/apps/gent/.../.../software/worker/.../bin/wsub: line 27: 2152510 <b>Illegal instruction</b>     (core dumped) ${PERL} ${DIR}/../lib/wsub.pl "$@"

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+/apps/gent/RHEL8/skylake-ib/software/worker/1.6.13-iimpi-2021b/bin/wsub: line 27: 2152510 <b>Illegal instruction</b>     (core dumped) ${PERL} ${DIR}/../lib/wsub.pl "$@"
+</code></pre>
+
+When executing the `module swap cluster/…` command, you are not only changing your environment to submit to the other cluster, but also to use the part of the central software stack that is specific to that cluster. The latter means that you are running a worker on top of a Perl installation that is optimized specifically for the CPUs in that cluster workernodes. That installation is not compatib ...
```
```suggestion
When executing the `module swap cluster` command, you are not only changing your session environment to submit
to that specific cluster, but also to use the part of the central software stack that is specific to that cluster.
In the case of the Worker example above, the latter implies that you are running the `wsub` command
on top of a Perl installation that is optimized specifically for the CPUs of the workernodes of that cluster,
which may not be compatible with the CPUs of the login nodes, triggering the "`Illegal instruction`" error.

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+
+When executing the `module swap cluster/…` command, you are not only changing your environment to submit to the other cluster, but also to use the part of the central software stack that is specific to that cluster. The latter means that you are running a worker on top of a Perl installation that is optimized specifically for the CPUs in that cluster workernodes. That installation is not compatib ...
+
+The cluster modules are split up into several "submodules" to help deal with this problem.
```
```suggestion
The cluster modules are split up into several `env/*` "submodules" to help deal with this problem.

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+When executing the `module swap cluster/…` command, you are not only changing your environment to submit to the other cluster, but also to use the part of the central software stack that is specific to that cluster. The latter means that you are running a worker on top of a Perl installation that is optimized specifically for the CPUs in that cluster workernodes. That installation is not compatib ...
+
+The cluster modules are split up into several "submodules" to help deal with this problem.
+By using `module swap env/slurm/<submit cluster>` instead of `module swap cluster/<submit cluster>` (starting from the default environment where the default cluster is loaded), you can update your environment to submit jobs to the `<submit cluster>`, while still using the software installations that are specific to the default cluster (which are compatible with the login nodes since the default c ...
```
```suggestion
For example, by using `module swap env/slurm/donphan` instead of `module swap cluster/donphan` (starting from the default environment where the default cluster is loaded), you can update your environment to submit jobs to `donphan`, while still using the software installations that are specific to the default cluster (which are compatible with the login nodes since the default cluster workernodes have the same CPUs).
The same goes for the other clusters as well, of course.

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+
+
+!!! Example
+    if you want to submit a worker job on a different cluster, please use
```
```suggestion
    If you want to submit a Worker job to the `donphan`, please use:
   ...
```

and adjust accordingly below, don't use `<submit cluster>` to avoid that people effectively use `<` and `>` in the command...

boegel commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,27 @@ The following have been reloaded with a version change:
+    instead of
+    <pre><code>$ <b>module swap cluster/<submit cluster\></b> </code></pre>
+
+We recommend using a `module swap cluster/<cluster>` command after submitting. This to "reset" your environment to a sane state, since only having a different `env/slurm/...` module loaded can also lead to some surprises if you're not paying close attention.
```
```suggestion
We recommend using a `module swap cluster` command after submitting the jobs.

This to "reset" your environment to a sane state, since only having a different `env/slurm` module loaded can also lead to some surprises if you're not paying close attention.

milachae commented on mkdocs/docs/HPC/multi_job_submission.md:
```diff
@@ -138,6 +138,14 @@ program will now be run for all 100 parameter instances -- 8
 a parameter instance is called a work item in Worker parlance.
 
+!!! warning
+    When you want to submit a job on a cluster that is not the default cluster. You need to use:
```
A better explanation has been added.

milachae commented:
@boegel 
I have added the requested changes.

I also added the macro for the default cluster, to make the text easier to read.

I also saw that there is a "othercluster" macro that is set to the donphan cluster, but I'm hesitant to use that one, due to the fact that it is not 'required' to be set to a debug/interactive cluster. The links to the interactive/debug cluster would make no sense if someone would change the othercluster to for example 'skitty'. 
There is also a possibility to add a new macro for the "debug/interactive cluster if we don't want to hard code into the text.

milachae commented on mkdocs/docs/HPC/troubleshooting.md:
```diff
@@ -479,4 +479,35 @@ The following have been reloaded with a version change:
+
+We recommend using a `module swap cluster` command after submitting the jobs.
+
+This to "reset" your environment to a sane state, since only having a different `env/slurm` module loaded can also lead to some surprises if you're not paying close attention.
```
I changed the last sentence to this:

```
We recommend using a `module swap cluster` command after submitting the jobs. Additional information about this as well as more comprehensive details concerning the 'Illegal instruction' error can be accessed [here](troubleshooting.md#multi-job-submissions-on-a-non-default-cluster).
```
